sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartATests.cs

```diff
@@ -46,99 +48,125 @@ public void InitRoleInfo_NullResource()
-        public void GeneratePartAEnvelope_DefaultActivity()
+        public void GeneratePartAEnvelope_DefaultActivity_DefaultResource()
         {
             var activity = CreateTestActivity();
-            var telemetryItem = TelemetryPartA.GetTelemetryItem(activity, null);
+            var resource = CreateTestResource();
+
+            var telemetryItem = TelemetryPartA.GetTelemetryItem(activity, resource, null);
+
             Assert.Equal("RemoteDependency", telemetryItem.Name);
             Assert.Equal(activity.StartTimeUtc.ToString(CultureInfo.InvariantCulture), telemetryItem.Time);
-            Assert.Null(telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()]);
+            Assert.StartsWith("unknown_service", telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()]);
             Assert.Null(telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()]);
             Assert.NotNull(telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()]);
             Assert.NotNull(telemetryItem.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()]);
             Assert.Throws<KeyNotFoundException>(() => telemetryItem.Tags[ContextTagKeys.AiOperationParentId.ToString()]);
         }
 
         [Fact]
-        public void GeneratePartAEnvelope_ActivityWithRoleInformation()
+        public void GeneratePartAEnvelope_Activity_WithResource()
         {
-            var activity = CreateTestActivity(
-                resource: Resources.CreateServiceResource("BusyWorker", "TEST3650724"));
+            var activity = CreateTestActivity();
+            var resource = CreateTestResource(serviceName: "my-service", serviceInstance: "my-instance");
+
+            var telemetryItem = TelemetryPartA.GetTelemetryItem(activity, resource, null);
 
-            var telemetryItem = TelemetryPartA.GetTelemetryItem(activity, null);
             Assert.Equal("RemoteDependency", telemetryItem.Name);
             Assert.Equal(activity.StartTimeUtc.ToString(CultureInfo.InvariantCulture), telemetryItem.Time);
-            Assert.Equal("BusyWorker", telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()]);
-            Assert.Equal("TEST3650724", telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()]);
+            Assert.Equal("my-service", telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()]);
+            Assert.Equal("my-instance", telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()]);
             Assert.Equal(activity.TraceId.ToHexString(), telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()]);
             Assert.Equal(SdkVersionUtils.SdkVersion, telemetryItem.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()]);
             Assert.Throws<KeyNotFoundException>(() => telemetryItem.Tags[ContextTagKeys.AiOperationParentId.ToString()]);
         }
 
         // TODO: GeneratePartAEnvelope_WithActivityParent
 
+        /// <summary>
+        /// If SERVICE.NAME is not defined, it will fall-back to "unknown_service".
```

Comment:
Is the concatenation with process.executable.name (prescribed in linked spec) something we target for a later PR?

Reply:
I ignored that field for these tests.
These tests are run for multiple frameworks and could run either locally hosted in Visual Studio or on a build server hosted by the command line.
Since the process name could change depending on how the tests are run i didn't think it was worth including.